### PR TITLE
Unvendor toml package

### PR DESCRIPTION
**Pull request by hoodmane**
Remove the vendored copy of `toml-j0.4` and use the version from `npm`.

### CODE REVIEW

**Comment by madhur-tandon** [comment]
Hi @hoodmane, IIRC, we specifically shipped it because we were looking for a TOML parser which is very lightweight i.e. the size of the library is very small. Further, I think we also modified it and added an additional check -- otherwise if we supply an invalid TOML, the parser hangs. OTOH, for the valid and correct TOML, there is no issue.

And I think, that is what is happening with the test failures also.
Since we added a test that specifically checks for an invalid config as well -- which fails here in this PR.

**Comment by hoodmane** [comment]
> very lightweight

Well it's a vendored copy of the npm toml library so using the unvendored version is going to be the same size.

> if we supply an invalid TOML, the parser hangs

Well it's especially bad if the vendored copy has been changed in some undocumented way... This sounds like an upstream bug, not some custom feature specific to our use case? Did someone open an issue on the toml library?

**Comment by hoodmane** [comment]
So the change is:
```
(PY1000): The config supplied: [[
 is an invalid TOML and cannot be parsed: SyntaxError: Expected "'", "\"", [ \t] or [A-Za-z0-9_\-] but "\n" found.
```
==>
```
(PY1000): The config supplied: [[
 is an invalid TOML and cannot be parsed: SyntaxError: Expected DoubleQuote, Whitespace, or [a-z], [A-Z], [0-9], "-", "_" but "\n" found
```
So the new error spells it out a bit more (but leaves out the option of SingleQuote). The package looks unmaintained so contributing an update to the error message seems not possible.

1. Leave the error message alone. It's not great, but it's better than the current situation.
2. Catch the error and substitute a better error message 
3. Fork the package and put in a better error message

Vendoring a copy of it is not a good option.

**Comment by hoodmane** [comment]
Ah well it seems we are using @ltd/j-toml which is maintained. It looks about 5x larger than toml though.

**Comment by madhur-tandon** [comment]
> Well it's a vendored copy of the npm toml library so using the unvendored version is going to be the same size.

I am not sure which one we came up with, maybe @tedpatrick remembers. I think it wasn't the official one, but maybe I have forgotten.

> Well it's especially bad if the vendored copy has been changed in some undocumented way... This sounds like an upstream bug, not some custom feature specific to our use case? Did someone open an issue on the toml library?

Yeah, this could be an upstream bug. I don't think there was an issue opened.

---

Actually, IIRC, we didn't fix the library, we just decided not to use the official one -- because that had the hanging issue on invalid TOMLs

And even with the other one, we had an additional check in our code. It is present here:

https://github.com/pyscript/pyscript/blob/main/pyscriptjs/src/pyconfig.ts#L149-L150

**Comment by hoodmane** [comment]
@iarna/toml gives an error like:
```
TomlError: Key ended without value at row 1, col 4, pos 3:
1> [[
      ^
```
which I think is perfectly clear.

File: pyscriptjs/src/pyconfig.ts [hunk 1]
```diff
@@ -145,17 +145,18 @@ function mergeConfig(inlineConfig: AppConfig, externalConfig: AppConfig): AppCon
 
 function parseConfig(configText: string, configType = 'toml') {
     if (configType === 'toml') {
+        // TOML parser is soft and can parse even JSON strings, this additional check prevents it.
```
Comment (madhur-tandon) [inline pyscriptjs/src/pyconfig.ts:148]: do we still need this additional check @hoodmane or the new library fixes this?

Comment (hoodmane) [reply pyscriptjs/src/pyconfig.ts:148]: I think it's the same library and the same checks are needed. We can try removing it in a followup but I would prefer to merge this as-is.

**Review by madhur-tandon — APPROVED** [review]

**Pull request merged**